core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3869,6 +3869,17 @@ private static AtomicLong getAtomicLong(String key) {
     return atomic;
   }
 
+  /** Support the ARRAY_COMPACT function. */
+  public static List removeNulls(List list) {
+    List result = new ArrayList();
+    for (int i = 0; i < list.size(); i++) {
+      if (list.get(i) != null) {
+        result.add(list.get(i));
+      }
+    }
+    return result;
+  }
+
   /** Support the ARRAY_DISTINCT function. */
```

Comment:
the reason why i don't use this because
1) the spark array(null) throw exception java.lang.UnsupportedOperationException: remove
    because the spark array implementation using Arrays.asList(); which make the exception happends
2) the standard array[null] pass
```
list.removeIf(Objects::isNull);
```

Reply:
Is there a reason to iterate by index rather than something like `for(Object obj : list)` ? Could help with readability a bit. Just an idea.

Reply:
fixed @tanclary